client/src/pages/PreGameLobby.js

```diff
@@ -11,32 +12,50 @@ import { GameContext } from "../context/GameContext";
 import FormInput from "../components/FormInput";
 import GenericButton from "../components/GenericButton";
 import useForm from "../utils/hooks/useForm";
-import Notification from '../components/Notification'
-import PlayerStatus from '../components/PlayerStatus'
-
+import Notification from "../components/Notification";
+import PlayerStatus from "../components/PlayerStatus";
+import sockets from "../utils/sockets";
 
 function PreGameLobby({ match }) {
   const queryGameId = match.params.gameId;
   const classes = useStyles();
   const { logoutUser } = useContext(AuthContext);
-  const { game, gameNotification, errors,
-    sendInvitation, joinGame, closeGameNotification, isCurrentUserHost } = useContext(GameContext);
+  const {
+    game,
+    gameNotification,
+    errors,
+    sendInvitation,
+    joinGame,
+    closeGameNotification,
+    isCurrentUserHost,
+  } = useContext(GameContext);
   const [playerEmail, setPlayerEmail] = useForm({ email: "" });
   const { players } = game;
+  const [gameStart, setGameStart] = useState(false);
 
   useEffect(() => {
     if (!game.gameId) {
       joinGame(queryGameId);
     }
+    sockets.on("game-started", () => {
+      setGameStart(true);
+    });
   }, [joinGame, queryGameId, game]);
 
   function startGame() {
-    console.log("start");
+    setGameStart(true);
+    sockets.emit("start-game", queryGameId);
+  }
+
+  if (isCurrentUserHost() && gameStart) {
+    return <Redirect to={`/session/${queryGameId}`} />;
+  } else if (!isCurrentUserHost() && gameStart) {
+    return <Redirect to={`/session/${queryGameId}`} />;
```

Comment:
Does it need to call isCurrentUserHost()? 
And this method occurs an error when a user creates a new game.
currentUser in isCurrentUserHost() is null, so it can't find the email property.

Reply:
Yeah, you're right. It doesn't need to call isCurrentUserHost(). Also, the currentUser property problem is part of something I commented on Insaf's previous PR. The next PR for this should fix both.